src/main/resources/assets/js/components/Marathon.jsx

```diff
@@ -159,12 +227,29 @@ var Marathon = React.createClass({
     }
   },
 
+  handleSetAppView: function (setAppView) {
```

Comment:
Yes, in an ideal case. Maybe we should handle this issue of passing things down in a seperate pull request.
See my comment here:
https://github.com/mesosphere/marathon/pull/1066#discussion-diff-23438066
This is a result of getting the route function out of the AppModalComponent.